File: app/src/main/java/com/commit451/frankenstein/sample/MainActivity.java
package com.commit451.frankenstein.sample;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

import java.util.concurrent.TimeUnit;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView textUsername = (TextView) findViewById(R.id.text_username);
        textUsername.setText(App.get().getUserName());
        findViewById(R.id.button_crash).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                throw new IllegalArgumentException("blah");
            }
        });
        findViewById(R.id.button_crash_delayed).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                v.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        throw new IllegalStateException("Blah");
                    }
                }, TimeUnit.SECONDS.toMillis(3));
            }
        });
    }
}

File: app/src/main/java/com/commit451/frankenstein/sample/MainActivity.kt
package com.commit451.frankenstein.sample

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

import java.util.concurrent.TimeUnit

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        textUsername.text = UserManager.username()
        buttonCrash.setOnClickListener { throw IllegalArgumentException("blah") }
        buttonCrashDelayed.setOnClickListener { v -> v.postDelayed({ throw IllegalStateException("Blah") }, TimeUnit.SECONDS.toMillis(3)) }
        buttonSecondActivity.setOnClickListener {
            val intent = Intent(this, SecondActivity::class.java)
            startActivity(intent)
        }
        switchShouldRelaunch.isChecked = App.get().shouldRelaunch
        switchShouldRelaunch.setOnCheckedChangeListener { _, isChecked ->
            App.get().shouldRelaunch = isChecked
        }
    }
}
